lib/src/component/http_req.rs

```diff
@@ -187,8 +181,17 @@ impl http_req::Host for Session {
             cursor,
         );
 
-        if buf.is_empty() && next.is_none() {
-            return Ok(None);
+        if buf.is_empty() {
+            if next.is_none() {
+                return Ok(None);
+            } else {
+                // It's an error if we couldn't write even a single value.
+                return Err(Error::BufferLengthError {
```

Comment:
Does this allow us to return the needed size? I'm guessing not, since this is not the `BufferLen` variant.

Reply:
Good point, I've fixed that by matching the behavior elsewhere: it returns the length needed to read at least one entry.

Reply:
This seems like a nice ergonomic improvement!